### PR TITLE
Tracking merge facility users in some connection breakage situations

### LINKED ISSUE #10268: User is not appropriately alerted to failures of the merge workflow when connection to the remote server is lost
Thanks @jredrejo, as always we are dealing with complex flows... I just tried again with the latest build assets and it's still not fully functioning correctly as the user is left guessing what is going on unless the console is being opened. Here are the main scenarios the way I see them:

Scenario 1: The target server is stopped before pressing the Continue button. In that case there's no message displayed to the user and only a `410 Gone` error in the console:

https://user-images.githubusercontent.com/79847249/225590784-529d6945-721e-4808-aaac-9051d82bd13c.mp4

Scenario 2: The target server is stopped after pressing the Continue button. In that case I do see the `Remotely preparing data: MorangoError` however nothing happens after restarting the target server and clicking the RETRY button - I have to manually refresh the page and start over in order to complete the process.

https://user-images.githubusercontent.com/79847249/225592996-1cfcd6d1-b930-4e50-8524-07861e23c7d3.mp4

Scenario 3: The target server is not being stopped but there are network interruptions before or after pressing the Continue button. In that case it's somewhat clearer to the learner what is going on however it also involves starting over the process.

https://user-images.githubusercontent.com/79847249/225603372-21de70cd-c9ca-45bb-88dc-50d4a4d59f2b.mp4

_Originally posted by @pcenov in https://github.com/learningequality/kolibri/issues/10247#issuecomment-1471786755_

### PR DESCRIPTION
**Pull request by jredrejo**
## Summary

Trying to add some solutions to connection failures when merging users in different situations, according to the scenarios @pcenov reported in #10268 

> Scenario 1: The target server is stopped before pressing the Continue button. In that case there's no message displayed to the user and only a 410 Gone error in the console

I'm afraid I'm not able to reproduce it, even following strictly the video, @pcenov  please, check if the error persists on your side. It might have been solved after your tests . I always get this screen when reproducing it:
error_merge1

> Scenario 2: The target server is stopped after pressing the Continue button. In that case I do see the Remotely preparing data: MorangoError however nothing happens after restarting the target server and clicking the RETRY button 

Reproduced and fixed here. However, I've been able to see, depending on the exact moment the remote server is closed, that sometimes the task stays forever and never reports as failure. I don't think that's something related to t his work, nor even to the syncs, but a problem in our current task system. Tasks should timeout at some point. If the task never reports a FAILED status, the local kolibri can't know there's a problem.

> Scenario 3: The target server is not being stopped but there are network interruptions before or after pressing the Continue button. In that case it's somewhat clearer to the learner what is going on however it also involves starting over the process.

Reproduced and fixed here.

## References
Closes: #10268 

## Reviewer guidance
Following the videos at #10268 try to reproduce those problems.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue [hunk 1]
```diff
@@ -143,67 +143,75 @@
       }
 
       function pollTask() {
+        console.log(taskId.value);
```
Comment (bjester) [inline kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue]: :eyes:

Comment (jredrejo) [reply kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeFacility.vue]: removed, thanks!

**Comment by pcenov** [comment]
Hi @jredrejo unfortunately I am still able to reproduce all 3 scenarios on my end so from a user point of view it's not clear to me what exactly has been fixed here. I'm not doing anything different than what I've already recorded in the videos. I've been testing using the following build from the Build Artefacts section: [kolibri_0.16.0a12.dev0+git.153.ga870d8c4-0ubuntu1_all.deb](https://github.com/learningequality/kolibri/suites/12277924228/artifacts/650326558)

Call me when you have time to further discuss.

**Comment by jredrejo** [comment]
> Hi @jredrejo unfortunately I am still able to reproduce all 3 scenarios on my end so from a user point of view it's not clear to me what exactly has been fixed here. I'm not doing anything different than what I've already recorded in the videos. I've been testing using the following build from the Build Artefacts section: [kolibri_0.16.0a12.dev0+git.153.ga870d8c4-0ubuntu1_all.deb](https://github.com/learningequality/kolibri/suites/12277924228/artifacts/650326558)
> 
> Call me when you have time to further discuss.

Hi @pcenov , will you have time to check this in a meeting this afternoon/evening? I have one meeting scheduled but I have time before and after the meeting to see what's happening here. I don't see these problems anymore, maybe it's a problem of the built artifact or there are some differences in our tests.

**Comment by jredrejo** [comment]
hi @pcenov finally, after being able to reproduce the error, we discovered the causes for the problem were deeper than expected and were caused by an undetected problem in the backend that now should be fixed. Once more time: thanks to your detailed reports and help.
Please, re-test it again and let me know if you still have any problem with the code.

**Comment by pcenov** [comment]
Hi @jredrejo after a lot of testing and regression testing I can confirm that the main cases are covered nicely now and the user attempting to change the facility gets notified in case of connection interruption.

At the same time I am reliably able to replicate an issue where if the target server gets stopped while locally integrating the received data I do see the message that the merging task has failed but when I restart the server and click the 'Retry' button I get into an infinite loop. If I refresh the page I can see that there are now 2 facilities on the machine and I can actually sign in to the target facility with the merged user.

See the following video with the errors in the console:

https://user-images.githubusercontent.com/79847249/234023787-0095436e-b545-4077-9fe7-90d19de88df9.mp4

Logs and DB:

[KolibriFolder.zip](https://github.com/learningequality/kolibri/files/11311374/KolibriFolder.zip)

Let me know if you think that this should be handled in this PR or we should close this one and create a follow up issue.

**Comment by jredrejo** [comment]
hi @pcenov 

> 
> Let me know if you think that this should be handled in this PR or we should close this one and create a follow up issue.

Honestly, I don't know the answer to your question because, one more time, I can not reproduce the problem.
Please, send me the also the db of the device that's merging, also, in order to try to guess what's  happening:
 - could you take a screenshot of the terminal messages where the server error trace should appear?
 - could you send me the db (or even better the kolibri folder) of the server where's the original user? you sent me the other one

Last but not least, trying to avoid making this an infinite issue, could you tell me all the cases you're testing so we can try to solve all the issues at once?. 
If that's not possible, I think it's better to create new issues and consider this one as finished (as all the problems in the description are fixed) because making this issue so long is avoiding to merge all the cases that have been fixed in the final kolibri release.

**Review by pcenov — APPROVED** [review]
Hi @jredrejo, I am no longer able to replicate the issue from my last comment and everything else is looking good in the latest build. I think we can close this one and merge it, and then if we stumble upon an edge case we'll open a separate issue to keep things neat.

File: kolibri/core/discovery/utils/network/client.py [hunk 2]
```diff
@@ -68,7 +69,10 @@ def build_for_address(cls, address, timeout=None):
             else:
                 # if we're within a request thread, then we limit it for an overall time
                 timeout = (DEFAULT_CONNECT_TIMEOUT, DEFAULT_SYNC_READ_TIMEOUT)
+        _, self_urls = get_urls()
         for url in get_normalized_url_variations(address):
+            if url in self_urls:
+                continue  # exclude our own URLs
```
Comment (bjester) [inline kolibri/core/discovery/utils/network/client.py:75]: :100:

**Review by bjester — APPROVED** [review]

**Pull request merged**